velox/serializers/PrestoSerializer.cpp

```diff
@@ -3354,6 +3354,18 @@ std::unique_ptr<VectorSerializer> PrestoVectorSerde::createSerializer(
       prestoOptions.compressionKind);
 }
 
+std::unique_ptr<VectorSerializer> PrestoVectorSerde::createSerializerFromRow(
```

Comment:
@kunalkataria I added some comments to the header on my last PR, check that out. The problem is that there are two ways to serialize data using this class. One where you create the serializer first, the use it to serialize many rowVectors (append()). In this case you cannot preserve the encoding as different rowVectors can have different encodings. 

Alternatively, you can use the serializeEncoded() method, in which you only serialize one rowVector, so you can keep the encodings, which is what you are trying to achieve.

Exposing this method presents a third method that is risky, since you want to preserve the encodings but are able to serialize multiple row vectors with potentially different encodings. 

Can't you use the static serializeEncoded() method directly?

Reply:
Ah I see - thanks for outlining that issue. I'll try changing the PrestoPage code to use serializeEncoded and see if that works